packages/Skipper/Fnmatcher.php

```diff
@@ -10,12 +10,41 @@ public function match(string $matchingPath, string $filePath): bool
     {
         $normalizedMatchingPath = $this->normalizePath($matchingPath);
         $normalizedFilePath = $this->normalizePath($filePath);
+
         if (\fnmatch($normalizedMatchingPath, $normalizedFilePath)) {
             return \true;
         }
 
         // in case of relative compare
-        return \fnmatch('*/' . $normalizedMatchingPath, $normalizedFilePath);
+        if (\fnmatch('*/' . $normalizedMatchingPath, $normalizedFilePath)) {
+            return true;
+        }
+
+        $realPathMatchingPath = realpath($normalizedMatchingPath);
+        $realpathNormalizedFilePath = realpath($normalizedFilePath);
+
+        if (! is_string($realPathMatchingPath)) {
+            return false;
+        }
+
+        if (! is_string($realpathNormalizedFilePath)) {
+            return false;
+        }
+
+        $realPathMatchingPath = $this->normalizePath($realPathMatchingPath);
+        $realpathNormalizedFilePath = $this->normalizePath($realpathNormalizedFilePath);
+
+        // skip define direct path
+        if (is_file($realPathMatchingPath)) {
+            return $realPathMatchingPath === $realpathNormalizedFilePath;
+        }
+
+        // ensure add / suffix to ensure no same prefix directory
+        if (is_dir($realPathMatchingPath)) {
+            $realPathMatchingPath = rtrim($realPathMatchingPath, '/') . '/';
+        }
+
+        return str_starts_with($realpathNormalizedFilePath, $realPathMatchingPath);
```

Comment:
The fnmatcher should handle only the * paths.
Rest shoule be managed by another matching case.

Reply:
Ok, I created `RealpatchMatcher` service for it https://github.com/rectorphp/rector-src/pull/2921/commits/30b4fcb76a43b4b55a9f69d61437b67250041994